docs/source/notes/configuration.md

```diff
@@ -0,0 +1,194 @@
+
+User needs to specify the dataset they are using by adding `dataset=<dataset_key>` option to their command.
+
+## Model Config
+
+Similar to dataset config, each model [registered](../lib/common/registry) to MMF can define its config. this is defined by model's `config_path` classmethod ([example](https://github.com/facebookresearch/mmf/blob/ae1689c0e2f9d8f51f337676495057168751c5ea/mmf/models/cnn_lstm.py#L40)). Configs for models live at [mmf/configs/models](https://github.com/facebookresearch/mmf/tree/ae1689c0e2f9d8f51f337676495057168751c5ea/mmf/configs/models). Again, like datasets models also provide some variations which can be used by including configs for those variations in the user config.
+
+User needs to specify the model they want to use by adding `model=<model_key>` option to their command. A sample model config would look like:
+
+```yaml
+model_config:
+    model_key:
+        random_module: ...
+```
+
+## User Config
+
+User can specify their configuration specific to an experiment or training setup by adding `config=<config_path>` argument to their command. User config can specify for e.g. training parameters according to their experiment such as batch size using `training.batch_size`. Most common use case for user config is to specify optimizer, scheduler and training parameters. Other than that user config can also include configs for variations of models and datasets they want to test on. Have a look at an example user config [here](../others/user_config).
+
+## Command Line Dot List Override
+
+Updating the configuration through [dot list](https://omegaconf.readthedocs.io/en/latest/usage.html#from-a-dot-list) syntax is very helpful when running multiple versions of an experiment without actually updating a config. For example, to override batch size from command line you can add `training.batch_size=x` at the end of your command. Similarly, for overriding an annotation in the hateful memes dataset, you can do `dataset_config.hateful_memes.annotations.train[0]=x`.
+
+```eval_rst
+
+.. note::
+
+    Command Line Dot List overrides are our recommended way of updating config parameters instead of manually updating them in config for every other change.
+```
+
+## Includes
+
+MMF's configuration system on top of OmegaConf allows building user configs by including composable configs provided by the datasets and models. You can include it following the syntax
+
+```yaml
+includes:
+- path/to/first/yaml/to/be/included.yaml
+- second.yaml
+```
+
+The configs will override in the sequence of how they appear in the directive. Finally, the config parameters defined in the current config will override what is present in the includes. So, for e.g.
+
+First file, `a.yaml`:
```

Comment:
In these examples can you use something that is related to our configs instead of mysql etc? Seems a bit odd.